### ISSUE TITLE
Deployment provides a 'Ready' status condition.

### ISSUE THREAD

**Issue by toumorokoshi** [kubernetes/kubernetes#101842]
#### What would you like to be added:

Ultimately I'm trying to figure out how I can use `kubectl wait` to wait for a deployment to complete. 

I don't believe that's possible today because [deployment only provides the following conditions](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment):

- `Available`, which tells you if the deployment can serve load.
- `Progressing` which tells you if the deployment is not erroring.
- `ReplicaFailure`, which tells you if some failure occurred while creating a replica.

Since [kubectl wait expects boolean condition flags as signal](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#wait), there's no way to extract some more fine-grained signal that could indicate readiness, such as maybe [NewReplicaSetAvailable](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go#L83). 

One future-proof condition looks to be the [Ready condition](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md#the-ready-condition), which cli-utils is looking to standardize on to output resource status.

#### Why is this needed:

`kubectl wait` looks to be the de-facto wait to wait on resources, which then enables some ordering or status guarantees in kubectl users. Not being able to wait until a deployment has completed means I can't order my deployment of applications (which I need for legacy reasons).

**Comment by neolit123** [comment]
duplicate of #95759

/close
/sig cli

**Comment by neolit123** [comment]
you can also ask users for tricks at the support channels:
https://github.com/kubernetes/kubernetes/blob/master/SUPPORT.md
